Kernel/Process.cpp

```diff
@@ -3285,6 +3286,12 @@ int Process::sys$getrandom(void* buffer, size_t buffer_size, unsigned int flags
     return 0;
 }
 
+int Process::sys$setkeymap(char* n_map, char* n_shift_map, char* n_alt_map)
+{
+    KeyboardDevice::the().set_maps(n_map, n_shift_map, n_alt_map);
```

Comment:
These pointers from userspace must be validated, otherwise a user process can trick the kernel into reading arbitrary memory.
Assuming it's `0x80` bytes per pointer:
```cpp
    if (!validate_read(n_map, 0x80))
        return -EFAULT;
    if (!validate_read(n_shift_map, 0x80))
        return -EFAULT;
    if (!validate_read(n_alt_map, 0x80))
        return -EFAULT;
```

Also, only the superuser should be allowed to do this, so let's fail with `-EPERM` if the current UID is non-zero:

```cpp
    if (!is_superuser())
        return -EPERM;
```